### PR TITLE
[FIX] account: fix cash rounding when rounding method is set to global

### PR DESCRIPTION
**Pull request by ushyme**
### Steps to reproduce
- install Accounting
- in the settings, set `Rounding Method` to `Round Globally` and activate `Cash Rounding`
- create a cash rounding with the `half-up` method, a `Add a rounding line` strategy and a rounding precision of  `0.05`
- create a 7.7% sales tax
- create an invoice and in the `Other Info` tab, set `Cash Rounding Method` to the rounding you just created
- add an invoice line with a price of 295.00 and the 7.7% tax

You should see that the generated rounding line is not correct. The total is 317.72; so we expect the rounding line to have a price of `-0.02`, to respect the cash rounding.

### Cause
The cash rounding computation is initiated here:
https://github.com/odoo/odoo/blob/1dcd071b27779e7d6d8f536c7dce7002d27212ba/addons/account/models/account_move.py#L944

However, `total_amount_currency` is not rounded. This introduces floating imprecision that end up messing with the calculation here: https://github.com/odoo/odoo/blob/563752358d233d4fc9cf1d7e7abd0b1656dc82c1/addons/account/models/account_cash_rounding.py#L53

If we take the example given in the reproduction steps, the values in that line will be as follows:
- `amout = -317.71500000000003`
- `self.round(amount) = -317.70000000000005`
- `difference = self.round(amount) - amount = 0.014999999999986358`

`difference` is then rounded to 0.01 which not correct.

### Note
This issue does not happen starting v16.0, as `total_amount_currency` is already rounded when `_compute_cash_rounding` is called.

opw-3133291

### CODE REVIEW

**Comment by ushyme** [comment]
@fw-bot up to 15.2

**Comment by ushyme** [comment]
@fw-bot up to saas-15.2

**Comment by ushyme** [comment]
Hello @h4818, could you take a look at this when you have some time ?

**Comment by ushyme** [comment]
@h4818

**Review by h4818 — APPROVED** [review]
This fix looks ok to me, however...

I am just wondering why we don't store the rounded value in the amount_currency field (as it is done in 16.0)
Also, it will be nice to have the test in 16.0 with a few more cases

@smetl

File: addons/account/tests/test_account_move_out_invoice.py [hunk 1]
```diff
@@ -1620,6 +1620,43 @@ def test_out_invoice_line_onchange_cash_rounding_1(self):
...
+            'invoice_cash_rounding_id': cash_rounding.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    'name': 'test product',
+                    'price_unit': 295.0,
+                    'tax_ids': [(6, 0, tax7_7.ids)],
+                }),
+            ],
+        })
+        self.assertEqual(invoice.amount_total, 317.70)
```
Comment (smetl) [inline addons/account/tests/test_account_move_out_invoice.py:1658]: use `assertRecordValues` instead

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -941,7 +941,7 @@ def _apply_cash_rounding(self, diff_balance, diff_amount_currency, cash_rounding
         others_lines -= existing_cash_rounding_line
         total_amount_currency = sum(others_lines.mapped('amount_currency'))
 
-        diff_balance, diff_amount_currency = _compute_cash_rounding(self, total_amount_currency)
+        diff_balance, diff_amount_currency = _compute_cash_rounding(self, self.currency_id.round(total_amount_currency))
```
Comment (smetl) [inline addons/account/models/account_move.py:944]: can you round the `amount_currency` in `_recompute_tax_lines` when creating the tax lines instead plz?

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**